### ISSUE TITLE
Extract Method refactoring

### ISSUE THREAD

**Issue by psfinaki** [dotnet/fsharp#14449]
I'm surprised I haven't found a ticket for that - basically I'd like this to work for Ctrl-R + Ctrl-M like in C#:

https://user-images.githubusercontent.com/5451366/206193451-38c7b7fa-701e-4853-81e8-7b2da2b49859.mp4

**Comment by vzarytovskii** [comment]
Yeah, extract to function and method (if in context of type) will be helpful
